### PR TITLE
[ACCSEC-18827] Sample app internal distribution

### PR DESCRIPTION
**Pull request by SergioFierro**
## Jira Ticket
- ACCSEC-18827

## Description
- Sample app internal distribution for testing for all environments (dev, stg, prod)
- Prod build will only be generated when a new releases is triggered in main branch
- Dev and stage builds will be generated in feature, release, hotfix, dev branches
- https://console.firebase.google.com/u/0/project/twilioverify-sampleapp/appdistribution/app/android:com.twilio.verify.sample/releases
- https://app.circleci.com/pipelines/github/twilio/twilio-verify-android/1039/workflows/0564b1dc-9384-4d6b-b96e-d4b9021c5680/jobs/3730

### CODE REVIEW

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: fastlane/Fastfile [hunk 1]
```diff
@@ -136,4 +139,41 @@ platform :android do
...
+        "baseURL" => options[:url],
+        "verifyVersionName" => options[:versionName]
+      },
+      task: "assemble",
+      build_type: options[:build_type],
+      print_command: false
+    )
+    path = lane_context[SharedValues::GRADLE_APK_OUTPUT_PATH]
+    firebase_app_distribution(
+      app: "1:116450802435:android:df6d5c263506bb20728736",
```
Comment (yafuquen) [inline fastlane/Fastfile:172]: Can we store this as a Circle CI env variable?

File: fastlane/Fastfile [hunk 2]
```diff
@@ -136,4 +139,41 @@ platform :android do
...
+      apk_path: path,
+      groups: "qa-team",
```
Comment (yafuquen) [inline fastlane/Fastfile:174]: Same than app

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by sanarroya — APPROVED** [review]

**Review by yafuquen — APPROVED** [review]

**Pull request merged**